tests/phpunit/tests/blocks/renderHooks.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Tests for the hooks of the block rendering functions.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.4.0
+ *
+ * @group blocks
+ */
+class Tests_Blocks_Render_Hooks extends WP_UnitTestCase {
+	/**
+	 * @ticket 59131
+	 */
+	public function test_do_blocks_pre_render_filter() {
+		$block_content = '
+			<!-- wp:core/paragraph --><p>Hello</p><!-- /wp:core/paragraph -->
+			<!-- wp:core/paragraph --><p>WordPress!</p><!-- /wp:core/paragraph -->
+		';
+
+		add_filter( 'do_blocks_pre_render', array( $this, 'do_blocks_pre_render_filter' ) );
```

Comment:
it would be nice if these had names relating to what they are doing to the test rather than names that explain where they go. that is, if I see `array( $this, 'remove_blocks_not_containing_wordpress' )` then I don't have to jump around looking for the filter function and then work backwards. same for the `post_` filter.

this is not a blocking point, but a nice to have

Reply:
It's also possible inline the filter using local functions so it's colocated in case it won't be used in other places.

See the example in the codebase:

https://github.com/WordPress/wordpress-develop/blob/2acd0aca7845feda1c4ca4dd708ca4e8a72c422b/tests/phpunit/tests/blocks/register.php#L930C1-L937

Reply:
Makes sense, thanks. Done in https://github.com/WordPress/wordpress-develop/pull/5023/commits/230a9ade032f1d2f3db85951c221903ef0445428.

Reply:
> It's also possible inline the filter using local functions so it's colocated in case it won't be used in other places.

Oh, ok. I didn't see that comment.

Done in https://github.com/WordPress/wordpress-develop/pull/5023/commits/5abfc9c4a2c81b82bdb489211db2493517e5b26b.